src/main/java/net/openhft/hashing/CharSequenceAccess.java

```diff
@@ -37,22 +37,55 @@ private static int ix(long offset) {
         return (int) (offset >> 1);
     }
 
-    static long getLong(CharSequence input, long offset,
-                        int char0Off, int char1Off, int char2Off, int char3Off) {
-        int base = ix(offset);
-        long char0 = input.charAt(base + char0Off);
-        long char1 = input.charAt(base + char1Off);
-        long char2 = input.charAt(base + char2Off);
-        long char3 = input.charAt(base + char3Off);
-        return char0 | (char1 << 16) | (char2 << 32) | (char3 << 48);
+    protected static long getLong(CharSequence input, long offset,
+                                  int char0Off, int char1Off, int char2Off, int char3Off,
+                                  int char4Off, int delta) {
+        final int base = ix(offset);
+        if (0 == ((int)offset & 1)) {
+            final long char0 = input.charAt(base + char0Off);
+            final long char1 = input.charAt(base + char1Off);
+            final long char2 = input.charAt(base + char2Off);
+            final long char3 = input.charAt(base + char3Off);
+            return char0 | (char1 << 16) | (char2 << 32) | (char3 << 48);
+        } else {
+            final long char0 = input.charAt(base + char0Off + delta) >>> 8;
+            final long char1 = input.charAt(base + char1Off + delta);
+            final long char2 = input.charAt(base + char2Off + delta);
+            final long char3 = input.charAt(base + char3Off + delta);
+            final long char4 = input.charAt(base + char4Off);
```

Comment:
Note that you must not query a single byte outside of the range: e. g. a sequence of 9 bytes (`new byte[9]`) and you query long at offset 1 - you must not try to pull 10 bytes (as you apparently do now) because the 10th byte may be outside of the allowed area.

Reply:
`CharSequenceAccess` delegate all memory read operations to `CharSequence.charAt()`, so the boundary of querying could at `char` level, not `byte` level. Suppose we could use `CharSeqenceAccess` to wrap a `b=byte[9]` array, then `access.getByte(b, 8)` (this method is not changed in this pr) will also read beyond the byte boundary, that is, `input.charAt(9>>2)` will eventually query `b[8]` and `b[9]`.

imo, this is a flaw from the birthday of CharSequenceAccess, so we have to not use this access implementation to query an odd byte length memory. luckily, this should be always true for `java.lang.CharSequence`.

Reply:
You are right, I missed that we are accessing CharSequences anyway